### ISSUE TITLE
Inquiry About KL Divergence in ICML 2024 Paper

### ISSUE THREAD

**Issue by junkangwu** [Vance0124/Token-level-Direct-Preference-Optimization#8]
Hi Yongcheng,
 
I hope this message finds you well.
 
I have a question regarding the KL divergence presented in Figure 3(a) of your ICML 2024 paper on Token-level Direct Preference Optimization. Could you please clarify whether the KL divergence is calculated using the test set results, or is it computed based on the divergence of y_w​ or y_l​ from the training set?
 
Thank you for your assistance. I look forward to your response.
 
Best regards,

**Comment by Vance0124** [comment]
Hello Junkang,

Yes, when calculating the KL divergence, we utilized the results from the test set. First, we generated responses corresponding to the prompts in the test set. Then, we scored these responses using the Siebert/sentiment-roberta-large-english model. Additionally, we computed the KL divergence between policy $\pi$ and the reference policy $\pi_{ref}$ based on these responses.

Best regards
